Automattic/wp-calypso
# Reduxify DnsStore in Domains

Migrate the Flux store for domain DNS records to a Redux one.

- remove the `needsDns` option from the `DomainManagementData` data provider component. The `DomainManagement.Dns` will declare its data needs in the `connect` wrapper and with `<Query*>` components, like the rest of Calypso. It also replaces the `needsDomains` flag with `<QuerySiteDomains>` et al. The `DomainManagementData` should eventually disappear.
- migrate the Flux store to a Redux one by moving the reducer, actions and selectors. That's quite straighforward, as the Flux store was already using a reducer.
- inline the `dns/utils` functions that only the reducer uses into the `reducer` module. Improves bundle splitting
- use the `Query` components and selectors for site domains and domain DNS in ``DomainManagement.Dns`. Requires a bit of refactoring to always render the `Query` components, both with and without placeholder.

**How to test:**
Edit DNS record for a domain, apply a DNS template for services like Office 365 or Google Apps. Verify that nothing is broken.
